src/containers/UserProfile.tsx

```diff
@@ -1,48 +1,69 @@
 import { useState } from "react";
 import Input from "../components/Input/Input";
 import Button from "../components/Button/Button";
-import fetchUsers from "../utils/fetchUsers";
+import fetchGithubData from "../utils/fetchGithubData.ts";
 import UserDetails from "../components/UserDetails/UserDetails.tsx";
+import UserRepos from "../components/UserRepos/UserRepos.tsx";
+import { Repo } from "../interface.ts";
 
 interface User {
   id: number;
-  valueInput?: string;
+  userLogin?: string;
+  login: string;
+  avatar_url: string;
+  name: string;
 }
 
 export default function UserProfile() {
   const [user, setUser] = useState<User>();
+  const [repos, setRepos] = useState<Repo[]>([]);
   const [error, setError] = useState<string | null>(null);
   const [inputValue, setInputValue] = useState<string>("");
 
-  async function fetchData(inputValue: string) {
+  async function fetchUser(inputValue: string) {
     try {
-      const data = await fetchUsers(inputValue);
+      const data = await fetchGithubData<User>(inputValue);
```

Comment:
this was better the way you had before.

`fetchUsers` and `fetchUserRepos` would be a better approach. Even though you might end up with a little bit of extra code, you should avoid as much as you can to have generic code that deals with domain data. This case you have one hook that delives Users and Repos, and this is not ideal, even less Ideal when we're talking about a typed system.